s4e-backend/src/main/java/pl/cyfronet/s4e/service/GeoServerService.java

```diff
@@ -31,8 +32,25 @@ public void resetWorkspace() {
         geoServerOperations.createWorkspace(geoServerProperties.getWorkspace());
     }
 
-    public void addLayer(Product product) {
-        // FIXME: implement
+    @Transactional(rollbackFor = RestClientResponseException.class)
+    public void addStoreAndLayer(Product product) {
+        // Both the coverage store, coverage and layer (the last one with workspace prefix) names will be the same
+        String gsName = product.getLayerName().toLowerCase();
```

Comment:
Do the lowercase thing when populating the field, and add it as a regex constrain on the field.